test/tools/libtesteth/ImportTest.cpp

```diff
@@ -283,8 +287,15 @@ std::tuple<eth::State, ImportTest::ExecOutput, eth::ChangeLog> ImportTest::execu
         ImportTest::checkBalance(_preState, initialState);
 
         //Finalize the state manually (clear logs)
-        bool removeEmptyAccounts = m_envInfo->number() >= se->chainParams().EIP158ForkBlock;
         initialState.commit(removeEmptyAccounts ? State::CommitBehaviour::RemoveEmptyAccounts : State::CommitBehaviour::KeepEmptyAccounts);
+
+        if (!removeEmptyAccounts)
+        {
+            // Touch here bacuse coinbase might be suicided above
+            initialState.addBalance(_env.author(), 0);  // touch coinbase no matter what
+            initialState.commit(removeEmptyAccounts ? State::CommitBehaviour::RemoveEmptyAccounts :
```

Comment:
What if we move this out of the `if` body?  Maybe we can skip the first `initialState.commit`?

Reply:
no. the first commit apply suicide instruction to coinbase. making it dissappear from the post state.

Reply:
thus simulating touch and commiting the state again.